### PR TITLE
support math functions in line_format and label_format

### LINKED ISSUE #2847: Adding Math functions to line_format
**Is your feature request related to a problem? Please describe.**
I have a scenario where AccessLogs from my reverse proxy have the duration logged as nanoseconds.
It would be great if I could format the extracted label "duration" as milliseconds (dividing it by 1000000).

**Describe the solution you'd like**
Having the Math and Conversion functions from sprig available on line_format as a follow-up of #2822 PR
Something like:

```
{log_type="access_log"}
| json
| line_format "{{.ClientHost}}\t{{.OriginStatus}}\t{{div (.Duration | int) 1000000}}\t{{.RequestScheme}}\t{{.RequestAddr}}\t{{.RequestPath}}"
```

**Describe alternatives you've considered**
Using Grafana to graph it as a metric

**Additional context**
http://masterminds.github.io/sprig/math.html
http://masterminds.github.io/sprig/conversion.html

### LINKED ISSUE #2955: Allow math on label_format expressions
I would suggest we open up `label_format` stages to allow mathematical operations. For instance, say you wanted counts based on status code groups (2xx, 4xx, etc).

I'd like to use something like this, taking into account integer division to do the grouping: ` | json | status="{{.status / 100}}"`

/cc @cyriltovena @chancez WDYT?

### PR DESCRIPTION
**Pull request by adityacs**
**What this PR does / why we need it**:
Add math functions to line_format and label_format.

**Which issue(s) this PR fixes**:
Fixes #2847, Fixes #2955 

**Checklist**
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Review by cyriltovena — COMMENTED** [review]
Code wise this looks good. I think though we need something better for the doc. Either we document all additions, or  we redirect to sprig by explaining what sub-set we support.

File: docs/sources/logql/template_functions.md [hunk 1]
```diff
@@ -371,3 +371,227 @@ Examples:
 {{ if .err hasSuffix "Timeout" }} timeout {{end}}
 {{ if hasPrefix "he" "hello" }} yes {{end}}
 ```
+
+## add 
+
+> **Note:** Added in Loki 2.2.
```
Comment (adityacs) [inline docs/sources/logql/template_functions.md:377]: Not sure if this should be `2.2`. We may have to change.

Comment (cyriltovena) [reply docs/sources/logql/template_functions.md:377]: Yeah this is going to be 2.3

**Review by cyriltovena — APPROVED** [review]
LGTM

**Pull request merged**